magicctapipe/scripts/lst1_magic/semi_automatic_scripts/coincident_events.py

```diff
@@ -0,0 +1,258 @@
+"""
+This scripts facilitates the usage of the script
+"lst1_magic_event_coincidence.py". This script is
+more like a "manager" that organizes the analysis
+process by:
+1) Creating the bash scripts for looking for
+coincidence events between MAGIC and LST in each
+night.
+2) Creating the subdirectories for the coincident
+event files.
+
+Usage:
+$ python coincident_events.py (-c config_file.yaml)
+"""
+import argparse
+import glob
+import logging
+import os
+from datetime import date as dtdt
+from datetime import timedelta
+from pathlib import Path
+
+import joblib
+import numpy as np
+import yaml
+
+from magicctapipe import __version__
+from magicctapipe.io import resource_file
+from magicctapipe.scripts.lst1_magic.semi_automatic_scripts.clusters import slurm_lines
+
+__all__ = ["configfile_coincidence", "linking_bash_lst"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def configfile_coincidence(ids, target_dir, source_name, NSB_match):
+
+    """
+    This function creates the configuration file needed for the event coincidence step
+
+    Parameters
+    ----------
+    ids : list
+        List of telescope IDs
+    target_dir : str
+        Path to the working directory
+    source_name : str
+        Name of the target source
+    NSB_match : bool
+        If real data are matched to pre-processed MCs or not
+    """
+
+    config_file = resource_file("config.yaml")
+    with open(
+        config_file, "rb"
+    ) as fc:  # "rb" mode opens the file in binary format for reading
+        config_dict = yaml.safe_load(fc)
+    coincidence = config_dict["event_coincidence"]
+
+    conf = {}
+    conf["event_coincidence"] = coincidence
+
+    if not NSB_match:
+        file_name = f"{target_dir}/{source_name}/config_coincidence.yaml"
+    else:
+        file_name = f"{target_dir}/v{__version__}/{source_name}/config_coincidence.yaml"
+    with open(file_name, "w") as f:
+        lines = [
+            "mc_tel_ids:",
+            f"\n    LST-1: {ids[0]}",
+            f"\n    LST-2: {ids[1]}",
+            f"\n    LST-3: {ids[2]}",
+            f"\n    LST-4: {ids[3]}",
+            f"\n    MAGIC-I: {ids[4]}",
+            f"\n    MAGIC-II: {ids[5]}",
+            "\n",
+        ]
+        f.writelines(lines)
+        yaml.dump(conf, f, default_flow_style=False)
+
+
+def linking_bash_lst(
+    target_dir, LST_runs, source_name, LST_version, env_name, NSB_match
```

Comment:
including again the comment because it got outdated due to a different change in the same line:
"you should pass also queueing system type to this function because you produce a bash file following slurm style"

Reply:
added